### PR TITLE
remove redundant config item name and default value from KeycloakConfig

### PR DESCRIPTION
**Pull request by machi1990**

### CODE REVIEW

File: extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java [hunk 1]
```diff
@@ -11,53 +11,53 @@
...
-    @ConfigItem(name = "realm")
+    @ConfigItem
     String realm;
 
     /**
-     * Name of the realm.
+     * Name of the realm key.
      */
     @ConfigItem(name = "realm-public-key")
     Optional<String> realmKey;
```
Comment (gsmet) [inline extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: Let's also try to be consistent here. Let's rename the property and remove the `name`.

File: extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java [hunk 2]
```diff
@@ -11,53 +11,53 @@
...
 
     /**
      * The client-id of the application. Each application has a client-id that is used to identify the application
      */
```
Comment (gsmet) [inline extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: The docs seems a bit off here. Or the name of the property. Or maybe it's just me :). Could you check what this one is used for?

File: extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java [hunk 3]
```diff
@@ -262,34 +262,34 @@
...
+    @ConfigItem
     boolean verifyTokenAudience;
 
     /**
      * If set to true will turn off processing of the access_token query parameter for bearer token processing.
      * Users will not be able to authenticate if they only pass in an access_token
      */
-    @ConfigItem(name = "ignore-oauth-query-parameter", defaultValue = "false")
+    @ConfigItem(name = "ignore-oauth-query-parameter")
     boolean ignoreOAuthQueryParameter;
```
Comment (gsmet) [inline extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: This one doesn't work and the property ends up being `o-auth`? I thought David had handled this case?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few questions and suggestions.

File: extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java (hunk 3 above)
Comment (machi1990) [reply extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: Yes exactly. It ends up being `o-auth`. I think only `JBoss` is handled differently.

File: extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java (hunk 2 above)
Comment (machi1990) [reply extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: Seems to be good. The doc is lifted from https://www.keycloak.org/docs/latest/securing_apps/index.html#_java_adapter_config

Comment (gsmet) [reply extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: OK. Not excited about it but I suppose it's better to be consistent with the JSON config file.

File: extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java (hunk 1 above)
Comment (gsmet) [reply extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: BTW, I'm talking about renaming the *Java* property.

Comment (machi1990) [reply extensions/keycloak/deployment/src/main/java/io/quarkus/keycloak/KeycloakConfig.java]: Yes. While I was at it I renamed several others config items.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**